src/core/ext/filters/load_reporting/server_load_reporting_filter.cc

```diff
@@ -242,11 +213,37 @@ void ServerLoadReportingCallData::RecvInitialMetadataReady(
   ServerLoadReportingChannelData* chand =
       reinterpret_cast<ServerLoadReportingChannelData*>(elem->channel_data);
   if (err == GRPC_ERROR_NONE) {
-    GRPC_LOG_IF_ERROR(
-        "server_load_reporting_filter",
-        grpc_metadata_batch_filter(calld->recv_initial_metadata_,
-                                   RecvInitialMetadataFilter, elem,
-                                   "recv_initial_metadata filtering error"));
+    if (const grpc_core::Slice* path =
+            calld->recv_initial_metadata_->get_pointer(
+                grpc_core::PathMetadata())) {
+      calld->service_method_ = path->Ref().TakeCSlice();
+    }
+    if (calld->target_host_ == nullptr) {
+      if (const grpc_core::Slice* authority =
+              calld->recv_initial_metadata_->get_pointer(
+                  grpc_core::AuthorityMetadata())) {
+        calld->target_host_len_ = authority->size();
+        calld->target_host_ =
+            reinterpret_cast<char*>(gpr_zalloc(calld->target_host_len_));
+        for (size_t i = 0; i < calld->target_host_len_; ++i) {
+          calld->target_host_[i] = static_cast<char>(
+              tolower(static_cast<unsigned char>(authority->begin()[i])));
+        }
+      }
+    }
+    std::string buffer;
+    auto lb_token = calld->recv_initial_metadata_->GetValue(
+        grpc_core::kGrpcLbLbTokenMetadataKey, &buffer);
+    if (lb_token.has_value()) {
+      if (calld->client_ip_and_lr_token_ == nullptr) {
+        calld->StoreClientIpAndLrToken(lb_token->data(), lb_token->size());
+      }
+      auto old = calld->recv_initial_metadata_->Remove(
```

Comment:
It would, but since this is not a callout and so hasn't been converted to the traits system, this is still using the older api's that don't have Take just yet.